### DISCUSSION TITLE
JFreeChart and Japanese

### DISCUSSION THREAD

**Discussion by ronandowd** [jfree/jfreechart#312] [Q&A]
Hi Guys,

I'm using JFreeChart (version `1.0.19`). We've been using it in our product for years all good (for english, i.e. `en-US` locale). We're adding in localization support into our product for Japanese (`ja-JP` locale).

I'm trying to get my line chart to show Japanese but I'm seeing as below.

![image](https://user-images.githubusercontent.com/1897383/188183142-ee58bb2e-d7a2-4398-93e5-4eebfbcad351.png)

The means I think that I need to import/use a TrueType font that can handle Japanese characters. I'm using a JBOSS server and do have my locale correctly set there to `ja-JP`

So I tried this but no joy.

                        DateAxis dateAxis = (DateAxis) plot.getDomainAxis();
                        axis.setLabelFont(myJapaneseFont);//$NON-NLS-L$
                        String fontPath = "C:\fonts\\";
                        String fontName = "meiryoreg.ttf"; //seems to be the font to use?
                        File file = new File(fontPath + fontName);
                        java.awt.Font myJapaneseFont = java.awt.Font.createFont(java.awt.Font.TRUETYPE_FONT, file);
                        myJapaneseFont = myJapaneseFont.deriveFont(java.awt.Font.PLAIN, 20f);
                        dateAxis.setLabelFont(myJapaneseFont); //this font is correct and looks in when I debug to here

In fact - even in `en-US` I can't change the size/details of any font which is really weird.

Would anyone have a link to any demos/code/samples where this has been done ?

Any help/guidance would be greatly appreciated. I can post my code samples too (just need to remove them from larger section of code and trim down for brevity here)

Thanks Ronan.

**Comment by trashgod** [comment]
This [Q&A](https://stackoverflow.com/q/26063828/230513) may be helpful, although it recounts findings on a different platform.

**Reply by ronandowd** [reply]
Thanks! - I'll take a look at that thread
